test/Gen.hs

```diff
@@ -125,52 +143,72 @@ class Forall g f | g -> f, f -> g where
 instance Forall (Rec '[]) (PropertyT IO ()) where
   forall' Nil = id
 
-instance (Forall (Rec gs) b, Show a) => Forall (Rec (Gen a ': gs)) (a -> b) where
+instance (Forall (Rec gs) b) => Forall (Rec (Gen a ': gs)) (a -> b) where
   forall' (g :. gs) f = do
-    a <- Hedgehog.forAll g
-    forall' gs (f a)
+    (a, labels) <- Hedgehog.forAll (runWriterT (runGen g))
+    traverse_ Hedgehog.label labels
```

Comment:
We now produce the labels for each generator and classify the property with them.